packages/frontend/src/utils/cache.js

```diff
@@ -0,0 +1,67 @@
+export default class Cache {
+    static CACHE_DB_NAME = 'Cache';
+
+    constructor(
+        version = 1,
+        storeName = 'unknown',
+        indexName = 'unknown'
+    ) {
+        this.dbVersion = version;
+        this.dbPromise = null;
+        this.storeName = storeName;
+        this.indexName = indexName;
+    }
+
+    onCreateScheme = Function.prototype;
+    onSuccess = Function.prototype;
+
+    _checkDbOpened() {
+        if (this.dbPromise) {
+            return;
+        }
+
+        throw Error('IndexedDB should be opened');
+    }
+
+    _createSchemeDelegate(open) {
+        return () => {
+            this.onCreateScheme(open);
+        };
+    }
+
+    async getObjectStore(access = 'readwrite') {
+        this._checkDbOpened();
+        const db = await this.dbPromise;
+        const tx = db.transaction(this.storeName, access);
+
+        return tx.objectStore(this.storeName);
+    }
+
+    async getIndexStore(access = 'readwrite') {
+        const store = await this.getObjectStore();
+
+        return store.index(this.indexName);
+    }
+
+    open() {
+        const open = indexedDB.open(Cache.CACHE_DB_NAME, this.dbVersion);
```

Comment:
`indexedDB` is propherty of `window`. Please, don't skip it and add -> `window.indexedDB.open(...`

Since it's `window` prop, anybody can overwrite it! So it's better to add check like:
```
if ('indexedDB' in window) {
  try {
     const request = window.indexedDB.open(DB_NAME, VERSION)
     etc...
  } catch (e) {
  }
}
```

Reply:
btw, i saw real situations from sentry logs on production, when indexeddb.open doesnt open an falls with error